datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_profile.py

```diff
@@ -0,0 +1,661 @@
+        profile_oid_collection[table_oid] = profile_node
+
+    metric_tags = []
+    for item in oid_node.indices:
+        mib = item.index_module
+        oid_name = item.index_name
+        # look for an alias
+        alias_key = '{}:{}'.format(mib, oid_name)
+        if aliases is not None and alias_key in aliases:
+            alias = aliases[alias_key]
+            mib = alias['MIB']
+            oid_name = alias['name']
+
+        index = {'MIB': mib, 'tag': oid_name}
+        column = {'name': oid_name}
+        index_oid = _find_oid_by_name(mib, oid_name, mibs_directories, json_mib_directory)
+        if index_oid is not None:
+            column['OID'] = index_oid
+            index_table_oid = '.'.join(index_oid.split('.')[:-2])
+            index_table_name = _find_name_by_oid(mib, index_table_oid, mibs_directories, json_mib_directory)
+            if index_table_name is not None:
+                index['table'] = index_table_name
+        index['column'] = column
+        metric_tags.append(index)
+
+    profile_oid_collection[table_oid]['metric_tags'] = metric_tags
+
+
+def _add_profile_column_node(profile_oid_collection, oid_node):
+    """
+    Updates a collection of profile nodes, indexed by oid, adding a column node
+    """
+    if oid_node.node_type != 'column':
+        raise ValueError('Must be a column node')
+    name = oid_node.name
+    # Table oids are defined as:
+    # <TABLE_OID>.<TABLE_ENTRY>.<COLUMN_NUM>
+    # where <TABLE_ENTRY> is always 1
+    oid = oid_node.oid
+    table_oid = '.'.join(oid.split('.')[:-2])
+    if table_oid not in profile_oid_collection:
+        # create table if it does not exist yet
+        table = {'OID': table_oid}
+        profile_node = {'table': table, 'symbols': [], 'metric_tags': []}
+        profile_oid_collection[table_oid] = profile_node
+
+    symbol = {'name': name, 'OID': oid}
+    if oid_node.description is not None:
+        symbol['description'] = oid_node.description
+    symbols = profile_oid_collection[table_oid]['symbols']
+    symbols.append(symbol)
+    profile_oid_collection[table_oid]['symbols'] = symbols
+
+
+def _add_profile_scalar_node(profile_oid_collection, oid_node):
+    """
+    Updates a collection of profile nodes, indexed by oid, adding a scalar node
+    """
+    if oid_node.node_type != 'scalar':
+        raise ValueError('Must be a scalar node')
+    name = oid_node.name
+    mib = oid_node.mib
+    oid = oid_node.oid
+
+    profile_node = {'MIB': mib}
+    if not oid.endswith('.0'):
+        oid = oid + '.0'
+    symbol = {'name': name, 'OID': oid}
+    if oid_node.description is not None:
+        symbol['description'] = oid_node.description
+    profile_node['symbol'] = symbol
+    profile_oid_collection[oid] = profile_node
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Generate an SNMP profile from a collection of MIB files')
+@click.argument('mib_files', nargs=-1)
```

Comment:
I think an error should be raised if no mib file is passed.

Currently we are getting this when no mib file is passed:

```
§ ddev meta snmp generate-profile-from-mibs
📋 Profile yaml dump copied to your clipboard
```